randovania/gui/multiplayer_session_window.py

```diff
@@ -578,6 +582,30 @@ async def duplicate_session(self):
         if new_name is not None:
             await self.game_session_api.duplicate_session(new_name)
 
+    def export_all_presets(self):
+        path = common_qt_lib.prompt_user_for_preset_folder(self)
+
+        if path is None:
+            return
+
+        illegal_chars_pattern = r"^[ .]|[/<>:\"\\|?*]+|[ .]$"
```

Comment:
you can use re.compile which should give a small performance boost

Reply:
Ended up using the already-existing `sanitize_for_path` function in `string_lib` to avoid duplication of efforts. It looks like it'll be SLOWER since it's iterating each character manually, but maybe *that* function could be optimized to use a regex some time in the future.